### PR TITLE
JP-2682: Discontinue use of drizpars reference file

### LINKED ISSUE #7080: Discontinue use of drizpars reference file in favor of resample pars reference file
_Issue [JP-2682](https://jira.stsci.edu/browse/JP-2682) was created on JIRA by [Alicia Canipe](https://jira.stsci.edu/secure/ViewProfile.jspa?name=acanipe):_

Based on discussion at the DMSWG meeting today about JP-2580 ([notes](https://outerspace.stsci.edu/display/JWSTPWG/2022-07-20+Meeting+notes%3A+DMSWG)), we would like to investigate the possibility of discontinuing the use of the drizpars reference file now that we have resample parameter files. Previously, the benefit of the drizpars file was that it allows multiple entries for a given observing mode, e.g., it can change the values of pixfrac depending on how many images are being combined. We can't do that with a parameter reference file; however, the CalWG made a decision to only use pixfrac=1, meaning we shouldn't need that functionality any more (right [Howard Bushouse](https://jira.stsci.edu/secure/ViewProfile.jspa?name=bushouse) ? Is that the only parameter with that feature being used?) This would simplify things for developers and the pipeline users.

[Anton Koekemoer](https://jira.stsci.edu/secure/ViewProfile.jspa?name=koekemoe) [Swara Ravindranath](https://jira.stsci.edu/secure/ViewProfile.jspa?name=swara) [Misty Cracraft](https://jira.stsci.edu/secure/ViewProfile.jspa?name=cracraft)  FYI

### PR DESCRIPTION
**Pull request by hbushouse**
Resolves [JP-2682](https://jira.stsci.edu/browse/JP-2682)

Closes #7080 

This PR removes all code from the resample/resample_spec steps that made use of the drizpars reference file. The parameter values will now either use the defaults set in the step code or the values provided in parameter reference files in CRDS. I retained the use of the "get_drizpars" function, which used to be load values from the drizpars ref file, but now it's been modified to simply construct the "kwargs" dictionary that gets passed to the drizzle method. All values for kwargs now come directly from the Step class, instead of the drizpars ref file. Also updated all the docs to remove references to drizpars.

**Checklist for PR authors (skip items if you don't have permissions or they are not applicable)**
- [x] added entry in `CHANGES.rst` within the relevant release section
- [ ] updated or added relevant tests
- [x] updated relevant documentation
- [x] added relevant milestone
- [x] added relevant label(s)
- [x] ran regression tests, post a link to the Jenkins job below.
      [How to run regression tests on a PR](https://github.com/spacetelescope/jwst/wiki/Running-Regression-Tests-Against-PR-Branches)
- [x] All comments are resolved
- [ ] Make sure the JIRA ticket is [resolved properly](https://github.com/spacetelescope/jwst/wiki/How-to-resolve-JIRA-issues)

### CODE REVIEW

**Comment by hbushouse** [comment]
So far I've tested many different modes by running tests locally. Will run a full regression test as soon as all truth files get updated for a recent MIRI ref file delivery (that leads to many expected differences).

File: jwst/resample/resample_step.py [hunk 1]
```diff
@@ -195,128 +161,42 @@ def _load_custom_wcs(asdf_wcs_file, output_shape):
 
         return wcs
 
-    def get_drizpars(self, ref_filename, input_models):
+    def get_drizpars(self, input_models):
```
Comment (braingram) [inline jwst/resample/resample_step.py:164]: Does this function use `input_models`?

**Review by melanieclarke — COMMENTED** [review]
Much cleaner!

It looks like there's a reference to the drizpars file in docs/jwst/resample/main.rst that should be updated, but other than that, I think this looks good.

File: jwst/resample/resample_step.py (hunk 1 above)
Comment (hbushouse) [reply jwst/resample/resample_step.py:164]: Hmmm, no it doesn't. Surprised style check didn't complain about that. I'll remove it. Good catch!

**Comment by hbushouse** [comment]
> Much cleaner!
> 
> It looks like there's a reference to the drizpars file in docs/jwst/resample/main.rst that should be updated, but other than that, I think this looks good.

Thanks, I missed that one, because it my "grep" missed the uppercase "DRIZPARS" that was used in the file. Fixed in latest commit.

File: jwst/resample/resample_step.py (hunk 1 above)
Comment (hbushouse) [reply jwst/resample/resample_step.py]: Fixed in latest commit.

**Comment by hbushouse** [comment]
Started a full regtest run at https://plwishmaster.stsci.edu:8081/job/RT/job/JWST-Developers-Pull-Requests/1512 with CRDS_PATH pointed to CRDS-TEST, in order to pickup and use resample/resample_spec pars ref files.

**Comment by hbushouse** [comment]
Regtest results show LOTS of differences, unfortunately many of them are simply due to the fact that a lot of routine ref file updates that are in CRDS-OPS are not in CRDS-TEST, and hence are unrelated to this PR. For those that are directly related, we see the "R_DRZPAR" keyword missing from output i2d and s2d products, because the drizpars ref file wasn't used, along with subtle changes in the image values, due to the use of weight_type="exptime" (instead of "ivm"). The value of the "RESWHT" keyword is also changed from "ivm" to "exptime" in these products.

**Comment by hbushouse** [comment]
I realized while reviewing the regtest results that we theoretically don't need the definition of the "R_DRZPAR" keyword anymore in the datamodel schema and hence it could be removed. But given the difficulties in coordinating updates in stdatamodels and jwst, it's easier if we just leave that as is for now (it doesn't hurt anything to be in there) and we remove it later in a separate action. I'll file a new ticket for that.

**Comment by melanieclarke** [comment]
> But given the difficulties in coordinating updates in stdatamodels and jwst, it's easier if we just leave that as is for now (it doesn't hurt anything to be in there) and we remove it later in a separate action. I'll file a new ticket for that.

You can use JP-3651 for any remaining clean up after this PR.

I also think it would be useful to see the regtests with the non-test CRDS, just to confirm things work okay when using the software defaults only.

**Comment by hbushouse** [comment]
OK, started another regtest run at https://plwishmaster.stsci.edu:8081/job/RT/job/JWST-Developers-Pull-Requests/1515 that uses CRDS-OPS. I'm not convinced, however, that this will give identical results to what's in our current truth files, because in the current version of resample, which still uses the drizpars ref file, some instruments/modes did have 1 or 2 params in the drizpars ref file that were set to non-default values. Given that the code in this PR won't load the drizpars ref files, it may run with 1 or 2 different param values. But I guess we'll see.

**Review by braingram — APPROVED** [review]
LGTM!

https://plwishmaster.stsci.edu:8081/job/RT/job/JWST-Developers-Pull-Requests/1515 looks to only have header differences. I'll have to give my eyes time to uncross after looking through all the "failures".

**Review by melanieclarke — APPROVED** [review]
Looks good to me too!

**Pull request merged**